TensorLib/NumpyRepr.lean

```diff
@@ -100,16 +113,123 @@ instance ByteArrayRepr : Repr ByteArray where
 structure NumpyRepr where
   header : NumpyHeader
   data : ByteArray
-  startIndex : Nat -- Pointer to the first byte of ndarray data
+  startIndex : Nat -- Pointer to the first byte of ndarray data, called `offset` in numpy
```

Comment:
Why are these not part of the header?

Reply:
I kept the header to what is actually there in the bytes, and added inferred/utility fields here. I'll document this better. To some extent it's arbitrary, though I think there will be multiple metadata objects pointing to the same underlying data so the easier we make that the better.

Reply:
I documented this better.